pwiz_tools/Skyline/Model/Prosit/Models/PrositModel.cs

```diff
@@ -359,35 +406,48 @@ private TPrositOut Predict(PredictionService.PredictionServiceClient predictionC
                 return null;
 
             // Make predictions batch by batch in sequence and merge the outputs
-            var prositOutputAll = new TPrositOut();
-            foreach (var prositIn in inputsList)
+            var prositOutput = new List<TPrositOut>();
+            for (int i = 0; i < inputsList.Count; ++i)
+                prositOutput.Add(null);
+
+            ParallelEx.For(0, inputsList.Count, batchIndex =>
             {
-                var prositOutput = Predict(predictionClient, prositIn, token);
-                prositOutputAll = prositOutputAll.MergeOutputs(prositOutput);
-
-                processed += prositIn.InputRows.Count;
-                progressStatus = progressStatus.ChangeMessage(TextUtil.SpaceSeparate(
-                        PrositResources.PrositModel_BatchPredict_Requesting_predictions_from_Prosit,
-                        processed.ToString(), @"/", totalCount.ToString()))
-                    .ChangePercentComplete(REQUESTING_INPUTS_FRACTION * processed / totalCount);
-                if (progressMonitor.UpdateProgress(progressStatus) == UpdateProgressResponse.cancel)
-                    return null;
-            }
+                var prositIn = inputsList[batchIndex];
+
+                if (progressMonitor.UpdateProgress(localProgressStatus) == UpdateProgressResponse.cancel)
+                    return;
+
+                prositOutput[batchIndex] = Predict(predictionClient, prositIn, token);
+
+                lock(progressMonitor)
+                {
+                    processed += prositIn.InputRows.Count;
+                    localProgressStatus = localProgressStatus.ChangeMessage(TextUtil.SpaceSeparate(
+                            PrositResources.PrositModel_BatchPredict_Requesting_predictions_from_Prosit,
+                            processed.ToString(), @"/", totalCount.ToString()))
+                        .ChangePercentComplete(REQUESTING_INPUTS_FRACTION * processed / totalCount);
+                    progressMonitor.UpdateProgress(localProgressStatus);
+                    Console.WriteLine(localProgressStatus.Message);
+                }
+            }, maxThreads: 4);
+
+            var prositOutputAll = prositOutput[0];

```

Comment:
I get ArgumentOutOfRangeException on this line when I have an empty Skyline document and I try to build a spectral library from Prosit.